src/RefractAST.cc

```diff
@@ -315,36 +327,82 @@ namespace drafter
                 }
             }
 
-            if (IElement* e = SimplifyRefractContainer(samples)) {
-                element->attributes["sample"] = e;
-            }
+        }
 
-            if (IElement* e = SimplifyRefractContainer(defaults)) {
-                element->attributes["default"] = e;
-            }
+        if (IElement* e = SimplifyRefractContainer(samples)) {
+            element->attributes["sample"] = e;
         }
 
-        if(!description.empty()) {
-            element->meta["description"] = IElement::Create(description);
+        if (IElement* e = SimplifyRefractContainer(defaults)) {
+            element->attributes["default"] = e;
         }
 
+
         return element;
     }
 
     template <typename T>
-    refract::IElement* RefractElementFromProperty(const mson::PropertyMember& property)
+    refract::MemberElement* RefractElementFromProperty(const mson::PropertyMember& property)
     {
-        refract::IElement* element = RefractElementFromValue<T>(property);
-        if (!property.name.literal.empty()) {
-            element->meta["name"] = refract::IElement::Create(property.name.literal);
+        //if (property.name.literal.empty()) {
+        //    throw std::logic_error("Propery has no set name");
+        //}
+
+        refract::MemberElement* element = new refract::MemberElement;
+        refract::IElement* value = RefractElementFromValue<T>(property);
+
+        if(!property.name.literal.empty()) {
+            element->set(property.name.literal, value);
         }
         else if (!property.name.variable.values.empty()) {
             if (property.name.variable.values.size() > 1) {
                 // FIXME: is there example for multiple variables?
                 throw std::logic_error("Multiple variables in property definition");
             }
-            element->meta["name"] = refract::IElement::Create(property.name.variable.values[0].literal);
-            element->attributes["variable"] = refract::IElement::Create(true);
+            element->set(property.name.variable.values.begin()->literal, value);
+            element->value.first->attributes["variable"] = refract::IElement::Create(true);
+            SetElementType(property.name.variable.typeDefinition, element->value.first);
+            // FIXME: 
+            // https://github.com/refractproject/refract-spec/blob/master/namespaces/mson-namespace.md#variable-property-name
```

Comment:
it is explained few lines below in same comment - line no 373
In spec is used "typeAttributes: sample"
But in discussion we propose to remove "sample && defalut" from "typeAttributes"

While i move this explaining text uder FIXME, will be OK?